File: backend/src/main/java/com/vehicle/maintenance/dto/reminder/ReminderResponse.java
package com.vehicle.maintenance.dto.reminder;

import com.vehicle.maintenance.enums.ReminderType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReminderResponse {

    private Long id;
    private Long vehicleId;
    private String vehicleInfo;
    private Long maintenanceId;
    private String title;
    private String description;
    private ReminderType reminderType;
    private Integer thresholdMileage;
    private LocalDate thresholdDate;
    private Boolean isRecurring;
    private Integer recurringInterval;
    private LocalDateTime lastNotified;
    private Boolean isActive;
    private LocalDateTime createdAt;
}

File: frontend/src/app/models/reminder/reminder-response.ts
export interface ReminderResponse {
  id: number;
  vehicleId: number;
  vehicleInfo?: string;
  maintenanceId?: number;
  title: string;
  description?: string;
  reminderType?: string;
  thresholdMileage?: number;
  thresholdDate?: string;
  isRecurring?: boolean;
  recurringInterval?: number;
  isActive: boolean;
  createdAt?: string;
}
